doc/maintenance.md

```diff
@@ -19,14 +17,12 @@ communication tools we use.
 
 - Added to the [pygmt-contributors team](https://github.com/orgs/GenericMappingTools/teams/pygmt-contributors) (gives 'write' permission to the repository)
 - Added as a collaborator on [DAGsHub](https://dagshub.com/GenericMappingTools/pygmt/settings/collaboration) (gives 'write' permission to dvc remote storage)
-- Added to the [PyGMT devs Slack channel](https://pygmtdevs.slack.com) (for casual conversations)
-- Added to the {doc}`Team Gallery page <team>`
```

Comment:
This entry is removed because, A new contributor usually cannot be recognized as "Distinguished Contributors" and is not qualified to be added to the "PyGMT team page".